packages/labs/tsserver-plugin/src/lib/rules/no-unassignable-property-bindings.ts

```diff
@@ -0,0 +1,144 @@
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {
+  getLitTemplateExpressions,
+  parseLitTemplate,
+} from '@lit-labs/analyzer/lib/lit/template.js';
+import {traverse, type Element} from '@parse5/tools';
+import {
+  PartType,
+  hasAttributePart,
+} from '@lit-labs/analyzer/lib/lit/template.js';
+import type * as ts from 'typescript';
+import {LitDiagnosticCode} from '../diagnostic-codes.js';
+import type {LitLanguageService} from '../lit-language-service.js';
+import {getLitExpressionType} from '../type-helpers/lit-expression-type.js';
+
+/**
+ * Type check property bindings. Gives an error if the property can't be found
+ * on the element, or if the expression is not assignable to the property type.
+ *
+ * Handles all the tricky cases, like `noChange`, `nothing`, and directives,
+ * provided that the directives are typed with enough information to infer
+ * the correct type of the directive class's `render` method.
+ */
+export const noUnassignablePropertyBindings = {
+  getSemanticDiagnostics(
+    sourceFile: ts.SourceFile,
+    typescript: typeof ts,
+    checker: ts.TypeChecker,
+    litLanguageService: LitLanguageService
+  ) {
+    const diagnostics: ts.Diagnostic[] = [];
+    const templates = getLitTemplateExpressions(
+      sourceFile,
+      typescript,
+      checker
+    );
+    for (const template of templates) {
+      const litTemplate = parseLitTemplate(template, typescript, checker);
+      const {tsNode} = litTemplate;
+
+      traverse(litTemplate, {
+        element(element: Element) {
+          for (const attr of element.attrs) {
+            if (!hasAttributePart(attr)) {
+              continue;
+            }
+            const part = attr.litPart;
+            if (part.type !== PartType.PROPERTY) {
+              continue;
+            }
+            let rightHandSideType: ts.Type;
+            // If there are more than two strings, or if either of the strings
+            // are not empty string, then the type of the right hand side is
+            // 'string'.
+            if (part.strings.length > 2 || part.strings.some((s) => s !== '')) {
+              rightHandSideType = checker.getStringType();
+            } else {
+              if (part.expressions.length !== 1) {
+                throw new Error(
+                  'InternalError: Expected exactly one expression'
+                );
+              }
+              const rhsExpression: ts.Expression = part.expressions[0];
+              rightHandSideType = getLitExpressionType(
+                checker.getTypeAtLocation(rhsExpression),
+                typescript,
+                litLanguageService.getProgram()!
+              );
+            }
+            // get the type of the left hand side
+            const elementType = litLanguageService.getElementClassType(
+              element.tagName
+            );
+            if (elementType === undefined) {
+              // Unknown element. Other rules will warn in this case.
```

Comment:
Does the HTMLElementTagNameMap include built-in elements? Should we fall back to HTMLElement for unknown elements? ie, a `.id=${}` binding is fine actually, but if we don't know the element `.foo=${}` is bad.